core-ui/src/components/Extensibility/components/CodeViewer.js

```diff
@@ -19,26 +19,38 @@ export function CodeViewer({ value, structure, schema }) {
   );
 }
 
-const getEditorValue = value => {
-  let parsedValue = '';
-  let language = '';
-
+function detectLanguage(value) {
   if (isValidYaml(value)) {
-    try {
-      parsedValue = jsyaml.dump(value);
-      language = 'yaml';
-    } catch (e) {
-      console.error(e);
-    }
+    return 'yaml';
   } else if (typeof value === 'object') {
-    try {
-      parsedValue = JSON.stringify(value, null, 2);
-      language = 'json';
-    } catch (e) {
-      console.error(e);
-    }
+    return 'json';
   } else if (typeof value === 'string') {
-    parsedValue = value;
+    return '';
+  }
+}
+function stringifyIfObject(value) {
+  return typeof value !== 'string' ? JSON.stringify(value, null, 2) : value;
+}
+
+const getValueAndLang = (value, structure) => {
+  let language = structure.language || detectLanguage(value);
+  let parsedValue = '';
+
+  try {
+    switch (language) {
+      case 'yaml':
+        parsedValue = jsyaml.dump(value);
+        break;
+      default:
+        //this includes JSON and other languages
+        parsedValue = stringifyIfObject(value);
+    }
+  } catch (e) {
+    window.alert(
```

Comment:
Do we really want to show a window alert for it? Could we just show a regular notification instead?

Reply:
Fixed